space-plugins/nuxt-base/nuxt.config.ts

```diff
@@ -9,4 +12,26 @@ export default defineNuxtConfig({
 	imports: {
 		dirs: ['./types', './utils'],
 	},
+  vite: {
+		plugins: [runTunnelPlugin()],
+  },
 });
+
+function runTunnelPlugin(): Plugin {
+	let isBuildingClientSideOfNuxt = false;
+	const tunnelCommand = process.env.TUNNEL_COMMAND;
+	return {
+		name: 'run-tunnel',
+		apply: 'serve',
+		config(config) {
+			isBuildingClientSideOfNuxt = config.build?.ssr !== true;
+		},
+		buildStart() {
+			if (tunnelCommand && isBuildingClientSideOfNuxt) {
```

Comment:
I'm not 100% knowledgeable on this, but AFAIK, Nuxt runs two different processes, one for client side, and another one for server side. To avoid duplicated tunnels, I'm just running the tunnel only from the the client side's process.